packages/trade-widget/src/components/AccountBlockie/Blockie.tsx

```diff
@@ -0,0 +1,196 @@
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+
+interface IdenticonPropTypes {
+    seed: string;
+    size?: number;
+    scale?: number;
+    color?: string;
+    bgColor?: string;
+    spotColor?: string;
+    [key: string]: any;
+}
+
+/* tslint:disable:member-access prefer-function-over-method */
+class Blockie extends React.Component<IdenticonPropTypes> {
+    public identicon: any;
+    constructor(props: IdenticonPropTypes) {
+        super(props);
+        this.generateIdenticon = this.generateIdenticon.bind(this);
+    }
+
+    componentDidMount() {
+        this.generateIdenticon({ ...this.props });
+    }
+
+    componentWillUpdate(nextProps: IdenticonPropTypes) {
+        if (!this.isEquivalent(this.props, nextProps)) {
+            this.generateIdenticon({ ...nextProps });
+        }
+    }
+
+    isEquivalent(prevProps: IdenticonPropTypes, nextProps: IdenticonPropTypes) {
+        const aProps = Object.getOwnPropertyNames(prevProps);
+        const bProps = Object.getOwnPropertyNames(nextProps);
+
+        if (aProps.length !== bProps.length) {
+            return false;
+        }
+
+        // tslint:disable-next-line:prefer-for-of
+        for (let i = 0; i < aProps.length; i++) {
+            const propName = aProps[i];
+
+            if (prevProps[propName] !== nextProps[propName]) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    generateIdenticon(options: any) {
+        // NOTE --  Majority of this code is referenced from: https://github.com/alexvandesande/blockies
+        //          Mostly to ensure congruence to Ethereum Mist's Identicons
```

Comment:
Why not just use `https://github.com/0xProject/0x-monorepo/blob/development/packages/website/ts/components/ui/identicon.tsx`?